### DISCUSSION TITLE
Windows: Tasmota dev. 14.1.0.4 under VSC 1.92.1 error in compiling

### DISCUSSION THREAD

**Discussion by HofeBY** [arendst/Tasmota#21941] [Support request] [answered]
### PROBLEM DESCRIPTION
Tasmota dev. 14.1.0.4 under fresh installed VSC 1.92.1 ended with Compiler Error of missing "as.exe", but this "as.exe" does exist.
Path lenght error ?

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x ] Device used (e.g., Sonoff Basic): _____
- [x ] Tasmota binary firmware version number used: dev. 14.1.0.4
  - [ ] Pre-compiled
  - [x ] Self-compiled
- [0 ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
I uninstall plattformIO, VSC, deleting %homepath%\\.platformio; deleting D:\Tasmota
Computer reboot
Installing fresh downloaded VSC 1.92.1, starting VSC, installing fresh platformIO v3.3.3, restart wished VSC.
downloading dev. TM14.1.0.4 as Zipfile, fresh unpacking to D:\Tasmota (like several times before)
cheking Internetconnection. OK   !
starting VSC and opening folder D:\Tasmota
NO configuration changes !
after updating process of VSC, Build start "tasmota32c6" and ended in Compiler error missing "... as.exe".

Compiler LOG.

/*
  tasmota_version.h - Version header file for Tasmota

  Copyright (C) 2021  Theo Arends

  This program is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.

  This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

#ifndef _TASMOTA_VERSION_H_
#define _TASMOTA_VERSION_H_

#define TASMOTA_SHA_SHORT                      // Filled by Github sed

const uint32_t TASMOTA_VERSION = 0x0E010004;   //  *  Executing task: C:\Users\Station\.platformio\penv\Scripts\platformio.exe run --environment tasmota32c6 

Processing tasmota32c6 (board: esp32c6; framework: arduino; platform: https://github.com/tasmota/platform-espressif32/releases/download/2024.08.10/platform-espressif32.zip)
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------Tool Manager: Installing https://github.com/tasmota/esptool/releases/download/v4.7.6/esptool.zip
Unpacking  [####################################]  100%
Tool Manager: tool-esptoolpy@4.7.6 has been installed!
Verbose mode can be enabled via `-v, --verbose` option
Nothing more to solidify
-DCONFIG_TASMOTA_FLASHMODE_QIO
*** use provided user_config_override.h as planned ***
*** use provided platformio_override.ini as planned ***
*** use provided platformio_tasmota_cenv.ini as planned ***
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32c6.html
PLATFORM: Espressif 32 (2024.8.10) > Espressif Generic ESP32-C6 >= 4M Flash, Tasmota 2880k Code/OTA, 320k FS
HARDWARE: ESP32C6 160MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-builtin, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 3.0.4+sha.0732d1b
 - tool-esptoolpy @ 4.7.6
 - tool-mklittlefs @ 3.2.0
 - tool-riscv32-esp-elf-gdb @ 11.2.0+20220823
 - tool-xtensa-esp-elf-gdb @ 11.2.0+20230208
 - toolchain-riscv32-esp @ 12.2.0+20230208
Converting tasmota.ino
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ strict
Found 125 compatible libraries
Scanning dependencies...
Dependency Graph
|-- Adafruit CCS811 Library @ 1.0.0
|-- Adafruit MAX31865 library @ 1.0.1
|-- Adafruit MCP9808 Library @ 1.1.2
|-- Adafruit PM25 AQI Sensor @ 1.0.6
|-- Adafruit seesaw Library @ 1.3.1
|-- Adafruit SGP30 Sensor @ 1.2.0
|-- Adafruit SGP40 Sensor @ 1.1.0
|-- Adafruit VEML7700 Library @ 1.0.0
|-- ESP8266Audio @ 1.9.5
|-- Berry @ 7.0
|-- Berry animation library for WS2812 leds @ 0.1
|-- Berry custom template @ 0.1
|-- Berry Matter protocol implementation @ 0.1
|-- Berry Tasmota mapping @ 1.0
|-- Berry int64 implementation for 32 bits architecture @ 1.0
|-- HASPmota @ 1.0
|-- Berry mapping to C @ 1.0
|-- BME8563
|-- headers
|-- ccronexpr
|-- WebServer @ 3.0.4
|-- EEPROM 24C128_256_521
|-- ESP32-to-ESP8266-compat @ 0.0.2
|-- FFat @ 3.0.4
|-- FS @ 3.0.4
|-- ams @ 1.0
|-- HttpClient light @ 1.0
|-- I2Cdevlib-Core
|-- JPEGDEC @ 1.2.7
|-- LibTeleinfo @ 1.1.7
|-- lvgl @ 9.1.0
|-- Berry mapping to LVGL @ 1.0
|-- arduino-mcp2515-1.0.1
|-- MPU6050 I2C
|-- Mutichannel_Gas_Sensor @ 0.0.1
|-- esp-nimble-cpp @ 1.4.1
|-- NTP Library
|-- OpenTherm Library @ 0.9.0
|-- Process_control @ 1.0.0
|-- re1.5 @ 0.9
|-- ScioSense ENS16x @ 8.0.0
|-- ScioSense ENS210 @ 3.0.0
|-- Sensirion I2C SGP41 @ 0.1.0
|-- SevenSegmentTM1637 @ 1.1.0
|-- SPL06-007 @ 0.1.0
|-- ESP Mail Client @ 3.4.9
|-- LVGL_assets @ 7.0
|-- UdpListener @ 1.0
|-- Unishox Compressor Decompressor highly customized and optimized for ESP8266 and Tasmota @ 1.0
|-- VL53L0X @ 1.0.2
|-- VL53L1X @ 1.0.1
|-- TLS mini client derived from Arduino @ 1.0
|-- Zip-readonly-FS @ 1.0
|-- A4988_Stepper @ 0.0.1
|-- Adafruit Fingerprint Sensor Library @ 2.1.0
|-- Adafruit GFX Library @ 1.5.6
|-- Adafruit LED Backpack Library @ 1.1.6
|-- Adafruit TSL2591
|-- base64 @ 1.1.1
|-- BME68x
|-- DHT sensor library @ 1.4.3
|-- DNSServer @ 3.0.4
|-- DS2480 Library @ 0.1.0
|-- Waveshare esp 2.9 inch e-paper display driver @ 1.0
|-- Waveshare esp 4.2 inch e-paper display driver @ 1.0
|-- Display renderer @ 1.0
|-- ESP KNX IP Library @ 0.5.2
|-- ESP8266SAM @ 1.0.1
|-- ESPFtpServer @ 0.1.0
|-- Ethernet @ 3.0.4
|-- Ext-printf @ 1.0
|-- FrogmoreScd30
|-- FrogmoreScd40
|-- FT5206_Library @ 1.0.0
|-- GT911
|-- HPMA115S0 Arduino Library @ 1.0.0
|-- IRremoteESP8266 @ 2.8.6
|-- JSMN JSON parser customized and optimized for ESP8266 and Tasmota @ 1.0
|-- KeeloqLib @ 1.1
|-- mp3_shine_esp32 @ 1.0.0
|-- LedControl @ 1.0.6
|-- LinkedList
|-- LiquidCrystal_I2C
|-- LittleFS @ 3.0.4
|-- TasmotaLList @ 1.0
|-- LOLIN_HP303B @ 1.0.0
|-- LoRa @ 0.8.0
|-- MFRC522 @ 1.4.7
|-- MLX90640
|-- MPU_accel
|-- NeoPixelBus @ 2.6.7
|-- NewPing @ 1.9.1
|-- OneWire @ 2.3.2
|-- PubSubClient @ 2.8
|-- RA8876 @ 1.0.2
|-- RadioLib @ 6.4.2
|-- rc-switch @ 1.0.0
|-- RF24 @ 1.3.3
|-- SD @ 3.0.4
|-- SD_MMC @ 3.0.4
|-- Sensirion I2C SEN5X @ 0.2.0
|-- SPI @ 3.0.4
|-- TasmotaModbus @ 3.6.0
|-- TasmotaSerial @ 3.6.0
|-- Ticker @ 3.0.4
|-- TM1638plus @ 1.7.0
|-- Joba_Tsl2561 @ 2.0.10
|-- BearSSL @ 0.6
|-- universal display Library @ 0.1
|-- WiFiHelper @ 1.0.0
|-- WiFi @ 3.0.4
|-- Wire @ 3.0.4
|-- XPT2046_Touchscreen
Building in release mode
Compiling .pio\build\tasmota32c6\src\tasmota.ino.cpp.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\code_signs.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\control_SNR.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\control_audio_bandwidth.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\control_codec.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\debug.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\dec_API.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\decode_core.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\decode_frame.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\decode_indices.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\decode_parameters.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\decode_pitch.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\decode_pulses.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\decoder_set_fs.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\enc_API.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\encode_indices.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\encode_pulses.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\LTP_analysis_filter_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\LTP_scale_ctrl_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\apply_sine_window_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\autocorr_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\burg_modified_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\corrMatrix_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\encode_frame_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\find_LPC_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\find_LTP_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\find_pitch_lags_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\find_pred_coefs_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\k2a_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\k2a_Q16_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\noise_shape_analysis_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\pitch_analysis_core_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\process_gains_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\regularize_correlations_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\residual_energy16_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\residual_energy_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\schur64_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\schur_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\vector_ops_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\fixed\warped_autocorrelation_FIX.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\gain_quant.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\init_decoder.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\init_encoder.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\inner_prod_aligned.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\interpolate.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\lin2log.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\log2lin.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\pitch_est_tables.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\process_NLSFs.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\quant_LTP_gains.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\resampler.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\resampler_down2.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\resampler_down2_3.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\resampler_private_AR2.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\resampler_private_IIR_FIR.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\resampler_private_down_FIR.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\resampler_private_up2_HQ.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\resampler_rom.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\shell_coder.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\sigm_Q15.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\sort.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\stereo_LR_to_MS.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\stereo_MS_to_LR.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\stereo_decode_pred.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\stereo_encode_pred.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\stereo_find_predictor.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\stereo_quant_pred.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\sum_sqr_shift.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\table_LSF_cos.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\tables_LTP.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\tables_NLSF_CB_NB_MB.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\tables_NLSF_CB_WB.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\tables_gain.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\tables_other.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\tables_pitch_lag.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\libopus\silk\tables_pulses_per_block.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\opusfile\info.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\opusfile\internal.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\opusfile\opusfile.c.o
Compiling .pio\build\tasmota32c6\lib8b6\ESP8266Audio\opusfile\stream.c.o
Compiling .pio\build\tasmota32c6\lib82d\berry_custom\be_custom_module.c.o
Compiling .pio\build\tasmota32c6\lib874\re1.5\backtrack.c.o
Compiling .pio\build\tasmota32c6\lib874\re1.5\charclass.c.o
Compiling .pio\build\tasmota32c6\lib874\re1.5\cleanmarks.c.o
Compiling .pio\build\tasmota32c6\lib874\re1.5\compile.c.o
Compiling .pio\build\tasmota32c6\lib874\re1.5\compilecode.c.o
Compiling .pio\build\tasmota32c6\lib874\re1.5\dumpcode.c.o
Compiling .pio\build\tasmota32c6\lib874\re1.5\pike.c.o
Compiling .pio\build\tasmota32c6\lib874\re1.5\recursive.c.o
Compiling .pio\build\tasmota32c6\lib874\re1.5\recursiveloop.c.o
Archiving .pio\build\tasmota32c6\lib82d\libberry_custom.a
riscv32-esp-elf-g++: fatal error: cannot execute 'c:/users/station/.platformio/packages/toolchain-riscv32-esp/bin/../lib/gcc/riscv32-esp-elf/12.2.0/../../../../riscv32-esp-elf/bin/as.exe': CreateProcess: No such file or directory
compilation terminated.
*** [.pio\build\tasmota32c6\src\tasmota.ino.cpp.o] Error 1
====================================================================================== [FAILED] Took 70.28 seconds ======================================================================================

Environment    Status    Duration
-------------  --------  ------------
tasmota32c6    FAILED    00:01:10.284
================================================================================= 1 failed, 0 succeeded in 00:01:10.284 ================================================================================= 

 *  The terminal process "C:\Users\Station\.platformio\penv\Scripts\platformio.exe 'run', '--environment', 'tasmota32c6'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 

#endif  // _TASMOTA_VERSION_H_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._
![File_as_exe](https://github.com/user-attachments/assets/93a8fb2c-3371-4f8f-ab7c-0b893435add2)

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
Yes, Windows problem only. Path length limits of ~250 chars. **Not** related to Tasmota
You can try to add in `platformio_override.ini` to workaround

```
[platformio]
core_dir = c:\.platformio
workspace_dir = c:\.pio
```

**Reply by HofeBY** [reply]
Thx for the tip & fast response.
It solves the problem !
It would have been nice to have read something about the installation requirements in the release notice for Tasmota on Windows under VSC.
It would be good to use it as standard under Windows with VSC, since it is obviously already prepared and only needs to be released.

**Comment by Wintespe** [comment]
Find out if you have a path length problem on Windows:

https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry

 Solutions are also described there.

**Comment by Jason2866** [comment]
Could be max. length of a command line too. Next "nice" limitation of Windows. Especially an issue when linking. Afaik there is no solution for.

**Reply by HofeBY** [reply]
Possibly yes.
But does such a path really make sense?
c:/users/station/.platformio/packages/toolchain-riscv32-esp/bin/../lib/gcc/riscv32-esp-elf/12.2.0/../../../../riscv32-esp-elf/bin/as.exe
Because if you look at the result, it is much shorter and possibly used without any errors.
c:/users/station/.platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/bin/as.exe

I also can't see any \ between the username "Station" and .platformio.
C:\Users\Station.platformio\penv\Scripts\platformio.exe

**Reply by barbudor** [reply]
That missing / between user name and '.platformio' is likely your problem
I haven't tried a Tasmota built on windows since a while but so far I never have such problem. I'll try in the next couple of days

**Comment by Wintespe** [comment]
?
https://learn.microsoft.com/en-us/troubleshoot/windows-client/shell-experience/command-line-string-limitation

**Comment by arendst** [comment, marked as answer]
I had the same issue recently. The only solution that worked for me was moving the .platformio folder from C:\User\Me to C:

As Windows should be able to support 255 chars (or 65000 with a registry fix) I cannot define why this fails from time to time.

**Reply by HofeBY** [reply]
Thx for the feedback, it's not just me who is experiencing this.
I understand the reason for your short username.
Had this for the first time with TM14.1.0.3

**Comment by Wintespe** [comment]
The PlatformIOdioe may have restrictions. This would require a manifest to be defined by the vendor. The compiler, linker, etc. are also possible. VisualStudioCode itself is from Microsoft and should have no restrictions.

**Comment by Jason2866** [comment]
As said Windows bullsh*t. All this trickery needed **only** for Windows.

**Comment by Wintespe** [comment]
What tricks do you mean? As you can read, Windows itself doesn't have any real problems. The default settings are set up for the average user to be as compatible as possible with older software. The settings can be easily tweaked. However, when software tool developers don't play by the rules and use outdated interfaces for Windows, it's a cause for concern.

**Comment by Jason2866** [comment]
Well every other OS does work without any issue. Special setups are only needed for Windows. The only real fix would be Windows solves this limitations without needing workarounds to overcome this Windows only problem.
Honestly why should developers take care of Windows bugs not solved correctly for maaany years.

**Comment by Wintespe** [comment]
Let's be honest, Windows bugs have been properly fixed for many years. It's just that the developers don't care about it.

**Reply by s-hadinger** [reply]
I have no clue what you mean by that. There is a limitation mentioned above that exists only on Windows (it's not even a bug). It makes it harder to use on Windows tools designed for Unix. 
Are you blaming Unix developers to not care about fixed bugs in Windows?

**Reply by Wintespe** [reply]
Of course, I don't want to blame  developerss. If you develop software for a specific platform, you should study the specifics of that environment and not attract unnecessary comparisons with other platforms. However, If you want to port it to other platforms, you may have to think outside the box.

**Comment by Basti149** [comment]
I have the same problem when compiling my ESp32 C2. The solution is a little triki. Isn't there an easier solution?

**Comment by Jason2866** [comment]
All bigger projects do run in this issue. Nice document from EspEasy (great project too) how to use WSL
https://espeasy.readthedocs.io/en/latest/Participate/PlatformIO.html#vs-code-with-wsl2

**Comment by nikito7** [comment]
To allow bigger paths it's needed ```\\?\C:``` ... but don't work.

For now I just use gitpod free account
